registry/storage/vacuum.go

```diff
@@ -49,6 +49,37 @@ func (v Vacuum) RemoveBlob(dgst string) error {
 	return nil
 }
 
+// RemoveManifestTagReferences removes a tag reference from the filesystem
+func (v Vacuum) RemoveManifestTagReferences(name string, dgst digest.Digest, tag string) error {
```

Comment:
Just fold this into `RemoveManifest`. Basically, we are just removing all of the references to the manifest in one go.